src/Microsoft.Diagnostics.EventFlow.Inputs.ILogger/Logger/EventFlowLogger.cs

```diff
@@ -0,0 +1,111 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Internal;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs.ILogger.Logger
+{
+    internal class EventFlowLogger : Extensions.Logging.ILogger
+    {
+        private readonly string categoryName;
+        private readonly LoggerInput loggerInput;
+        private Scope<object> scope;
+
+        public EventFlowLogger(string categoryName, LoggerInput loggerInput)
+        {
+            Debug.Assert(categoryName != null);
```

Comment:
Understood, but even for internal types, I like to do parameter checking and throw if the checks fail. I like to think that public methods & properties form a contract with consumers and parameter checking is part of that contract. OK to just assert the correctness of parameters of private methods.